### ISSUE TITLE
Latest release (0.14.0) fails to build

### ISSUE THREAD

**Issue by kaj** [aisk/rust-memcache#115]
Adding memcache 0.14.0 as a dependency causes build to fail with lots of messages like "the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied", making it impossible to use memcache with the currently stable rust toolchain.

This seems to be a dependency compatibility problem.

<details><summary>Cargo toml</summary>

```toml
[package]
name = "t"
version = "0.1.0"
authors = ["Rasmus Kaj <kaj@kth.se>"]
edition = "2018"

[dependencies]
memcache = "0.14.0"

```
</details>

<details><summary>Build log</summary>

```
   Compiling memcache v0.14.0
error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
   | |     ------^-------------------------------------------------------------------------
   | |_____|_____|
   |       |     the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::auth`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
   | |     ------^-------------------------------------------------------------------------
   | |_____|_____|
   |       |     the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::auth`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
   | |     ---------^---------------------------------------------
   | |_____|________|
   |       |        the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::version`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
   | |     ---------^---------------------------------------------
   | |_____|________|
   |       |        the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::version`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
   | |     -------^-----------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::flush`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
   | |     -------^-----------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::flush`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
   | |     ------------------^-----------------------------------------------------
   | |_____|_________________|
   |       |                 the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::flush_with_delay`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
   | |     ------------------^-----------------------------------------------------
   | |_____|_________________|
   |       |                 the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::flush_with_delay`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
36 | |     fn get<V: FromMemcacheValueExt>(&mut self, key: &str) -> Result<Option<V>, MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::get`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
36 | |     fn get<V: FromMemcacheValueExt>(&mut self, key: &str) -> Result<Option<V>, MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::get`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
36 | |     fn get<V: FromMemcacheValueExt>(&mut self, key: &str) -> Result<Option<V>, MemcacheError>;
37 | |     fn gets<V: FromMemcacheValueExt>(&mut self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError>;
   | |        ---^ - required by this bound in `protocol::ProtocolTrait::gets`
   | |___________|
   |             the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
35 | |     fn flush_with_delay(&mut self, delay: u32) -> Result<(), MemcacheError>;
36 | |     fn get<V: FromMemcacheValueExt>(&mut self, key: &str) -> Result<Option<V>, MemcacheError>;
37 | |     fn gets<V: FromMemcacheValueExt>(&mut self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError>;
   | |        ---^ - required by this bound in `protocol::ProtocolTrait::gets`
   | |___________|
   |             the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
37 | |     fn gets<V: FromMemcacheValueExt>(&mut self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError>;
38 | |     fn set<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::set`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
37 | |     fn gets<V: FromMemcacheValueExt>(&mut self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError>;
38 | |     fn set<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::set`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
38 | |     fn set<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
39 | |     fn cas<V: ToMemcacheValue<Stream>>(
   | |        --^ - required by this bound in `protocol::ProtocolTrait::cas`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
38 | |     fn set<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
39 | |     fn cas<V: ToMemcacheValue<Stream>>(
   | |        --^ - required by this bound in `protocol::ProtocolTrait::cas`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
45 | |     ) -> Result<bool, MemcacheError>;
46 | |     fn add<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::add`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
45 | |     ) -> Result<bool, MemcacheError>;
46 | |     fn add<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
   | |        --^ - required by this bound in `protocol::ProtocolTrait::add`
   | |__________|
   |            the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
46 | |     fn add<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
47 | |     fn replace<V: ToMemcacheValue<Stream>>(
   | |        ------^ - required by this bound in `protocol::ProtocolTrait::replace`
   | |______________|
   |                the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
46 | |     fn add<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError>;
47 | |     fn replace<V: ToMemcacheValue<Stream>>(
   | |        ------^ - required by this bound in `protocol::ProtocolTrait::replace`
   | |______________|
   |                the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
52 | |     ) -> Result<(), MemcacheError>;
53 | |     fn append<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
   | |        -----^ - required by this bound in `protocol::ProtocolTrait::append`
   | |_____________|
   |               the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
52 | |     ) -> Result<(), MemcacheError>;
53 | |     fn append<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
   | |        -----^ - required by this bound in `protocol::ProtocolTrait::append`
   | |_____________|
   |               the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
53 | |     fn append<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
54 | |     fn prepend<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
   | |        ------^ - required by this bound in `protocol::ProtocolTrait::prepend`
   | |______________|
   |                the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
53 | |     fn append<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
54 | |     fn prepend<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
   | |        ------^ - required by this bound in `protocol::ProtocolTrait::prepend`
   | |______________|
   |                the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
54 | |     fn prepend<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
55 | |     fn delete(&mut self, key: &str) -> Result<bool, MemcacheError>;
   | |     --------^------------------------------------------------------
   | |_____|_______|
   |       |       the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::delete`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
54 | |     fn prepend<V: ToMemcacheValue<Stream>>(&mut self, key: &str, value: V) -> Result<(), MemcacheError>;
55 | |     fn delete(&mut self, key: &str) -> Result<bool, MemcacheError>;
   | |     --------^------------------------------------------------------
   | |_____|_______|
   |       |       the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::delete`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
55 | |     fn delete(&mut self, key: &str) -> Result<bool, MemcacheError>;
56 | |     fn increment(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
   | |     -----------^------------------------------------------------------------------
   | |_____|__________|
   |       |          the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::increment`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
55 | |     fn delete(&mut self, key: &str) -> Result<bool, MemcacheError>;
56 | |     fn increment(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
   | |     -----------^------------------------------------------------------------------
   | |_____|__________|
   |       |          the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::increment`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
56 | |     fn increment(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
57 | |     fn decrement(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
   | |     -----------^------------------------------------------------------------------
   | |_____|__________|
   |       |          the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::decrement`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
56 | |     fn increment(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
57 | |     fn decrement(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
   | |     -----------^------------------------------------------------------------------
   | |_____|__________|
   |       |          the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::decrement`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
57 | |     fn decrement(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
58 | |     fn touch(&mut self, key: &str, expiration: u32) -> Result<bool, MemcacheError>;
   | |     -------^-----------------------------------------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::touch`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
57 | |     fn decrement(&mut self, key: &str, amount: u64) -> Result<u64, MemcacheError>;
58 | |     fn touch(&mut self, key: &str, expiration: u32) -> Result<bool, MemcacheError>;
   | |     -------^-----------------------------------------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::touch`

error[E0277]: the trait bound `protocol::ascii::AsciiProtocol<stream::Stream>: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
58 | |     fn touch(&mut self, key: &str, expiration: u32) -> Result<bool, MemcacheError>;
59 | |     fn stats(&mut self) -> Result<Stats, MemcacheError>;
   | |     -------^--------------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::ascii::AsciiProtocol<stream::Stream>`
   |       required by `protocol::ProtocolTrait::stats`

error[E0277]: the trait bound `protocol::binary::BinaryProtocol: protocol::ProtocolTrait` is not satisfied
  --> /home/kaj/.cargo/registry/src/github.com-1ecc6299db9ec823/memcache-0.14.0/src/protocol/mod.rs:31:11
   |
31 |   pub trait ProtocolTrait {
   |  ___________^
32 | |     fn auth(&mut self, username: &str, password: &str) -> Result<(), MemcacheError>;
33 | |     fn version(&mut self) -> Result<String, MemcacheError>;
34 | |     fn flush(&mut self) -> Result<(), MemcacheError>;
...  |
58 | |     fn touch(&mut self, key: &str, expiration: u32) -> Result<bool, MemcacheError>;
59 | |     fn stats(&mut self) -> Result<Stats, MemcacheError>;
   | |     -------^--------------------------------------------
   | |_____|______|
   |       |      the trait `protocol::ProtocolTrait` is not implemented for `protocol::binary::BinaryProtocol`
   |       required by `protocol::ProtocolTrait::stats`

error: aborting due to 34 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `memcache`.

To learn more, run the command again with --verbose.

```
</details>

**Comment by kaj** [comment]
The reason seems to be that enum_dispatch 0.2.3 is not compatible with 0.2.2.

**Comment by antonok-edm** [comment]
This should be fixed as of `enum_dispatch` 0.2.4.

**Comment by kaj** [comment]
Yes, seems to work again now.  Thanks!
